src/assets/GLSLAsset.js

```diff
@@ -0,0 +1,62 @@
+const Asset = require('../Asset');
+const localRequire = require('../utils/localRequire');
+const path = require('path');
+
+class GLSLAsset extends Asset {
+  constructor(name, pkg, options) {
+    super(name, pkg, options);
+    this.type = 'js';
+  }
+
+  async parse() {
+    await localRequire('glsl-token-inject-block', this.name);
+    const glslify = await localRequire('glslify', this.name);
+
+    const basedir = path.dirname(this.name);
+    const glsl = glslify(this.contents, {basedir});
+
+    return new GLSLAst(glsl, {basedir});
+  }
+
+  async collectDependencies() {
+    const glslifyDeps = require('glslify-deps');
+    const depper = glslifyDeps({cwd: this.ast.basedir});
```

Comment:
Won't this cause us to read and parse the files twice: once above as part of `glslify` transform, and once again here? I wonder if we can only do it once.

Reply:
Yeah, unfortunately it will. I tried to find the way to do it only once, but `glslify` doesn't export the needed functions in order to split it up conveniently. Maybe we could accomplish this with copy-pasting some `glslify` code, but I think it's not desirable.
Anyway, I believe that scanning for dependencies twice won't be a huge overhead, since it doesn't really compile the files, but just traverses them with regexps.

Reply:
The actual compile process, that involves concatenating GLSL files and other stuff, will be performed only once